Mathlib/CategoryTheory/Enriched/Limits/HasConicalLimits.lean

```diff
@@ -86,27 +99,57 @@ section Results
 
 variable {J : Type u₁} [Category.{v₁} J] {J' : Type u₂} [Category.{v₂} J']
 variable (V : Type u') [Category.{v'} V] [MonoidalCategory V]
-variable {C : Type u} [Category.{v} C] [EnrichedOrdinaryCategory V C]
+variable (C : Type u) [Category.{v} C] [EnrichedOrdinaryCategory V C]
+
+namespace HasConicalLimit
+
+variable {C} (F : J ⥤ C) [HasConicalLimit V F]
 
 /-- ensure existence of a conical limit implies existence of a limit -/
-example (F : J ⥤ C) [HasConicalLimit V F] : HasLimit F := inferInstance
+example : HasLimit F := inferInstance
 
+variable {F} in
 /-- If a functor `F` has a conical limit, so does any naturally isomorphic functor. -/
-lemma HasConicalLimit.of_iso {F G : J ⥤ C} [HasConicalLimit V F] (e : F ≅ G) :
+lemma of_iso {G : J ⥤ C} (e : F ≅ G) :
     HasConicalLimit V G where
   toHasLimit := hasLimit_of_iso e
   preservesLimit_eCoyoneda X := preservesLimit_of_iso_diagram (eCoyoneda V X) e
 
-instance HasConicalLimit.of_equiv (F : J ⥤ C) [HasConicalLimit V F]
-    (G : J' ⥤ J) [G.IsEquivalence] : HasConicalLimit V (G ⋙ F) where
+instance of_equiv (G : J' ⥤ J) [G.IsEquivalence] : HasConicalLimit V (G ⋙ F) where
 
+omit [HasConicalLimit V F] in
 /-- If a `G ⋙ F` has a limit, and `G` is an equivalence, we can construct a limit of `F`. -/
-lemma HasConicalLimit.of_equiv_comp (F : J ⥤ C) (G : J' ⥤ J) [G.IsEquivalence]
+lemma of_equiv_comp (G : J' ⥤ J) [G.IsEquivalence]
     [HasConicalLimit V (G ⋙ F)] : HasConicalLimit V F :=
   have e : G.inv ⋙ G ⋙ F ≅ F := G.asEquivalence.invFunIdAssoc F
   HasConicalLimit.of_iso V e
 
-variable (C)
+/-- Use the axiom of choice to extract explicit `ConicalLimitCone F` from `HasConicalLimit F`. -/
+noncomputable def getConicalLimitCone : ConicalLimitCone V F where
+  limitCone := getLimitCone F
+  isConicalLimit X := Classical.choice <|
+    (preservesLimit_eCoyoneda X).preserves (getLimitCone F).isLimit
+
+/-- An arbitrary choice of conical limit cone for a functor. -/
+noncomputable def conicalLimitCone : ConicalLimitCone V F := getConicalLimitCone V F
+
+/-- An arbitrary choice of conical limit object of a functor. -/
+noncomputable def conicalLimit : C := (conicalLimitCone V F).limitCone.cone.pt
```

Comment:
I am afraid this is an unnecessary duplication of the code from the limits API. As I had suggested, we already have `limit F`. The idea is to rely on the assumption `HasConicalLimit` and the preservation of limits API in order to obtain properties of this `limit F`.

Then, as I had initially suggested, I think it would be better, given a `c : Cone F`, to only define a structure `IsConicalLimit V c` which extends `IsLimit c`, and this structure should essentially be used only as a constructor for `HasConicalLimit`.